dialog/fileicon.go

```diff
@@ -0,0 +1,88 @@
+package dialog
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+type fileIcon struct {
+	widget.BaseWidget
+
+	mimeType, mimeSubType string
+}
+
+func (i *fileIcon) CreateRenderer() fyne.WidgetRenderer {
+	var res fyne.Resource
+	switch i.mimeType {
+	case "application":
+		res = theme.FileApplicationIcon()
+	case "audio":
+		res = theme.FileAudioIcon()
+	case "image":
+		res = theme.FileImageIcon()
+	case "text":
+		res = theme.FileTextIcon()
+	case "video":
+		res = theme.FileVideoIcon()
+	default:
+		res = theme.FileIcon()
+	}
+	img := canvas.NewImageFromResource(res)
+	extText := canvas.NewText(i.mimeSubType, theme.BackgroundColor())
```

Comment:
I don't think that mimeSubType is the same as extension.
This is user facing so we should present what they are familiar with... the visible extension.